### PR TITLE
feat: Issue reminder job

### PR DESCRIPTION
**Pull request by Chigala**
## What does this PR do?

This PR creates a remainder job which basically sends a remainder to the user after 36hrs if the user hasn't opened a PR

Fixes # (issue)
Screenshot 2024-03-02 at 09 57 53

## How should this be tested?
You'll need trigger API Key and the API URL to test the issue reminder job. The API url is `https://api.trigger.dev`(if you are using trigger cloud). To learn more about this visit (https://trigger.dev/docs/documentation/introduction)

### Required

- [ ] Filled out the "How to test" section in this PR
- [x] Read [How we Code at oss.gg](<[https://github.com/formbricks/formbricks/blob/main/CONTRIBUTING.md](https://formbricks.com/docs/contributing/how-we-code)>)
- [X] Self-reviewed my own code
- [X] Commented on my code in hard-to-understand bits
- [X] Ran `pnpm build`
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [ ] Merged the latest changes from main onto my branch with `git pull origin main`
- [ ] My changes don't cause any responsiveness issues

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the Formbricks Docs if changes were necessary

### CODE REVIEW

File: jobs/issueReminder.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
+import { getOctokitInstance } from "@/lib/github/utils";
+import { client } from "@/trigger";
+import { eventTrigger } from "@trigger.dev/sdk";
+import { z } from "zod";
+
+client.defineJob({
+  // This is the unique identifier for your Job, it must be unique across all Jobs in your project.
+  id: "issue-remainder-job",
```
Comment (ShubhamPalriwala) [inline jobs/issueReminder.ts:8]: Can we also fix the typo from `remainder` to `reminder` across the app? 🙏🏼

File: .env.example [hunk 2]
```diff
@@ -41,3 +41,8 @@ GITHUB_APP_PRIVATE_KEY=""
 
 # GitHub App Name in your App's Installation URL (https://github.com/apps/${GITHUB_APP_SLUG}/installations/new)
 GITHUB_APP_SLUG=
+
+#Trigger API credential  URL(https://trigger.dev/docs/documentation/introduction)
+TRIGGER_API_KEY=
+TRIGGER_API_URL=
+NEXT_PUBLIC_TRIGGER_PUBLIC_API_KEY=
```
Comment (ShubhamPalriwala) [inline .env.example:48]: Do we need the `NEXT_PUBLIC_TRIGGER_PUBLIC_API_KEY`?

File: lib/github/utils.ts [hunk 3]
```diff
@@ -13,11 +14,19 @@ export const getOctokitInstance = (installationId: number) => {
...
   }
 
+  //Converting PKCS#1 to PKCS#8
+  //For it to work trigger it has to be in  PKCS#8 format
+
+  const privateKeyPkcs8 = crypto.createPrivateKey(GITHUB_APP_PRIVATE_KEY).export({
+    type: "pkcs8",
+    format: "pem",
+  });
+
```
[the remark this answers is not in the document]
Comment (Chigala) [reply lib/github/utils.ts:24]: If I revert this, it'll throw an error when the trigger job runs. I think that was why you weren't able to see the comment, I had the same issue while I was testing.

**Comment by Chigala** [comment]
Hello @ShubhamPalriwala sorry I totally forgot about this. I think the 2 events were created because of a bug from smee.io payload delivery, sometimes it appears to  call our api endpoint more than once, this in turn creates more than one trigger event. I noticed this while testing too

File: trigger.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import { TriggerClient } from "@trigger.dev/sdk";
+
+export const client = new TriggerClient({
+  id: "testing-bHhV",
```
[the remark this answers is not in the document]
Comment (Chigala) [reply trigger.ts:4]: Nope, it doesn't have to be an env variable.

File: .env.example (hunk 2 above)
Comment (Chigala) [reply .env.example:48]: We really don't need this for now

File: trigger.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (ShubhamPalriwala) [reply trigger.ts:5]: So if we create a new Trigger.dev account for production? This stays the same?

File: .env.example (hunk 2 above)
Comment (ShubhamPalriwala) [reply .env.example:48]: Can we remove this then across the app?

**Comment by ShubhamPalriwala** [comment]
Hey @Chigala, let me know once the PR is ready from your end then I'll test & review it again. Currently I see that you have added comments where you plan to improve the codebase. Looking forward to them 💪🏼

File: trigger.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (Chigala) [reply trigger.ts:5]: The endpoint id is just there to group all your endpoint url(staging, dev and prod) in the trigger.dev dashboard. You can change it to any name you want.

**Review by ShubhamPalriwala — APPROVED** [review]
Thanks for the changes! Just tested this & it works nicely with trigger.dev! 💪🏼 Solid work!

PS: I just made a small commit to revert back the change in the constants file for ossgg label!

**Pull request merged**